samples/apps/smallbank/app/smallbank.cpp

```diff
@@ -167,26 +173,33 @@ namespace ccfapp
         int value = params["value"];
 
         if (name.empty())
+        {
           return make_error(
             HTTP_STATUS_BAD_REQUEST, "A name must be specified");
+        }
 
         auto account_view = tx.get_view(tables.accounts);
         auto account_r = account_view->get(name);
 
         if (!account_r.has_value())
+        {
           return make_error(HTTP_STATUS_BAD_REQUEST, "Account does not exist");
+        }
 
         auto savings_view = tx.get_view(tables.savings);
         auto savings_r = savings_view->get(account_r.value());
 
         if (!savings_r.has_value())
+        {
           return make_error(
             HTTP_STATUS_BAD_REQUEST, "Savings account does not exist");
+        }
 
         if (savings_r.value() + value < 0)
         {
-          return make_error(
-            HTTP_STATUS_BAD_REQUEST, "Not enough money in savings account");
+          // We do not want to abort the entire tx, just not complete
```

Comment:
So there's an explicit flag to preserve writes on error now added by @eddyashton (set_apply_writes, #1054), pretty much exactly for this exact purpose: the transaction returns an error but should still be committed.